### ISSUE TITLE
Already closed connections showing up in the containers tab

### ISSUE THREAD

**Issue by iaguis** [weaveworks/scope#2181]
## Steps to reproduce

1. Start a busybox docker container and connect to the internet:

```
$ docker run -it busybox
# nc google.com 80
```

2. Wait until the edge `container -> internet` is shown in the containers tab.

3. Close the connection e.g. with Ctrl-C

4. Start a second busybox docker container and connect to the internet again:

```
$ docker run -it busybox
# nc google.com 80
```

5. See that the UI shows both containers connected to the internet

If I switch to another tab (e.g. Processes) and back to the Containers one the problem is fixed: only the connection from the second container shows up so I guess it's a problem in the UI layer.

**Comment by 2opremio** [comment]
@iaguis Does reloading the UI (i.e. reloading the browser page) make the connection go away?

**Comment by iaguis** [comment]
Yes, reloading the browser page makes the connection go away.

Even changing to another view and going back to "Containers" makes it go away as I say in the OP.

**Comment by 2opremio** [comment]
Then it must be a UI problem. Thanks for the confirmation.

**Comment by jpellizzari** [comment]
The root cause of this appears to be the way we are merging updates. The `adjacency` property is not present on the node object after a connection has been removed. When we call `.merge()`, it keeps the old one because the previous state has adjacency, and does not delete the adjacency key.

**Comment by jpellizzari** [comment]
Fixed by #2194
